bioblend/galaxy/jobs/__init__.py

```diff
@@ -191,6 +191,25 @@ def get_metrics(self, job_id):
         url = self._make_url(module_id=job_id) + '/metrics'
         return self._get(url=url)
 
+    def cancel_job(self, job_id):
+        """
+        creates a cancel request for a specified job
+
+        :type job_id: str
+        :param job_id: job ID
+
+        :rtype: requests.Response
+        :return: the response object.
+
+        .. note::
+         Calling ``get_state()`` to determine current job's state
+         if the state is not ok then make a delete_request
+        """
+
+        if self.get_state(job_id) != "ok":
+            url = '/'.join((self._make_url(job_id), "/discarded"))
```

Comment:
This API endpoint doesn't seem to exist as far as I can tell? It should be sufficient to make the DELETE request to `/api/jobs/{id}`

Reply:
as documented here: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/api/jobs.py#L244